### PR TITLE
enable choose term for collect title

### PR DESCRIPTION
**Pull request by wangben718**
Use argument `title` in `tlf_ae_*` function to enable user to control title collection.

### CODE REVIEW

**Review by LittleBeannie — APPROVED** [review]

File: R/tlf_ae_summary.R [hunk 1]
```diff
@@ -43,7 +43,7 @@ tlf_ae_summary <- function(outdata,
                            col_rel_width = NULL,
                            text_font_size = 9,
                            orientation = "portrait",
-                           title = NULL,
+                           title = c("analysis", "observation", "population"),
```
Comment (BrianLang) [inline R/tlf_ae_summary.R:46]: Is there a reason why we need to give 3 options to trigger this? 
What happens if someone passes in a character vector of 
`c("analysis", "this is a real title", "this is another real title")` and expect those to literally be the titles of their output?

File: R/tlf_ae_summary.R [hunk 2]
```diff
@@ -58,12 +58,13 @@ tlf_ae_summary <- function(outdata,
...
   # Define title
-  if (is.null(title)) {
+  if ("analysis" %in% title | "observation" %in% title | "population" %in% title) {
     title <- collect_title(outdata$meta,
       outdata$population,
       outdata$observation,
       parameters[1],
-      analysis = "ae_summary"
+      analysis = "ae_summary",
+      title_order = title
```
Comment (BrianLang) [inline R/tlf_ae_summary.R:67]: in the case where this `if` statement is satisfied, do you expect always all three "analysis", "observation" and "population" to be in the title vector?
It seems like you want something like `if(all(c("analysis", "observation", "population") %in% title))`

Comment (wangben718) [reply R/tlf_ae_summary.R:67]: Hi @BrianLang . Sometimes we don't need all these three component for title. For example, we only need "analysis" and "population". That is the main reason I want to update. 

If user really want have a title like `title = c("analysis", "this is a real title", "this is another real title")`, they could input `"analysis \n  this is a real title \n this is another real title"` instead. Maybe we could have more document for this issue to tell user "analysis", "observation", "population" are the special keyword we use? Also welcome to any other idea for the check the `title` argument input!

Comment (wangben718) [reply R/tlf_ae_summary.R:67]: I also thought about adding one more argument, such as `title_order` to control the title component. But I feel there might make user confused about the functionality of `title` and `title_order`.

File: R/tlf_ae_summary.R (hunk 1 above)
Comment (elong0527) [reply R/tlf_ae_summary.R:46]: Changing the default value may be related to my recent PR to extend the flexibility of defining title https://github.com/Merck/metalite/blob/main/R/collect.R#L286

We probably should change the argument to `title_order` to be aligned with `metalite`.

There should be a input checking to only allow component name to be used. 

The actual title should be defined while user define a `metalite` object and reused in different places.

File: R/tlf_ae_specific.R [hunk 3]
```diff
@@ -25,7 +25,7 @@
 #' @inheritParams r2rtf::rtf_page
 #' @inheritParams r2rtf::rtf_body
 #' @param footnotes A character vector of table footnotes.
-#' @param title A character vector of table titles.
+#' @param title Term "analysis", "observation"and "population") for collecting title from metadata or a character vector of table titles.
```
Comment (elong0527) [inline R/tlf_ae_specific.R:28]: you might want to separate the argument into two `title_order` and `title`. 

Each one is for different purpose. Then you can allow a vector of user defined `title` as @BrianLang suggest.

Comment (BrianLang) [reply R/tlf_ae_specific.R:28]: would it be:
- if user gives an argument for `titles`, then use those titles (and ignore `title_order`). 
- if `titles == NULL` then use `title_collect` and `title_order`. 
  - with argument checking for `title_order` of something like `sort(title_order) == sort(c("analysis", "observation", "population"))`? Or is it not necessary to have all three ?

Comment (elong0527) [reply R/tlf_ae_specific.R:28]: Exactly! The reason to have all three it to ensure flexibility in metalite. The title component can be any order of combination to fit use cases in different tables.

**Pull request merged**